### PR TITLE
ENH: add optional argument, "include_nyquist", for freqz()

### LINKED ISSUE #8265: freqz() does not include evaluation at pi by default
The default, or when specifying worN as an integer, the frequencies are evaluated at

```
linspace(0, 2*pi if whole else pi, N, endpoint=False)
```

However, I believe it would make more sense to *include* the endpoint, as the complex value of the frequency at pi is valuable information that I believe should be visible by default. This would parallel the often-taken step of taking a N-point FFT, and, for even N, keeping the values at indexes 0..N/2 *inclusive*. The last value represents the frequency at the Nyquist frequency.

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
To fix #8265

#### What does this implement/fix?
add optional argument, "nyquist", for freq()

### CODE REVIEW

File: scipy/signal/filter_design.py
Comment (tylerjereddy) [inline scipy/signal/filter_design.py]: Could maybe use a `.. versionadded:: 1.5.0` directive, but wait for Eric to review first.

**Review by tylerjereddy — COMMENTED** [review]
Looks reasonably solid at first glance---will have to let Eric check it though.

File: scipy/signal/filter_design.py
Comment (larsoner) [reply scipy/signal/filter_design.py]: Yes it will indeed need this directive.

Also `nyquist` should probably only be used/considered in the `linspace` call when `not whole`. Otherwise in `whole` mode you'll include 0 and 2*pi, which is probably not desired. So the docs would then look something like:
```suggestion
        If `whole` is False and `worN` is an integer, setting `nyquist` to True
        will include the last frequency (Nyquist frequency).
```

File: scipy/signal/filter_design.py
Comment (larsoner) [inline scipy/signal/filter_design.py]: As per the idea above:
```suggestion
        w = np.linspace(0, lastpoint, N, endpoint=nyquist and not whole)
```

File: scipy/signal/tests/test_filter_design.py
Comment (larsoner) [inline scipy/signal/tests/test_filter_design.py]: Then please also add a test that when `whole=True` you get the same result when `nyquist=False` as when `nyquist=True`

File: scipy/signal/filter_design.py
Comment (ilayn) [reply scipy/signal/filter_design.py]: I would also reserve nyquist keyword to something else. `endpoint` or something less serious would be better.

Comment (larsoner) [reply scipy/signal/filter_design.py]: `include_nyquist` perhaps? Because it's really about whether or not to include the Nyquist freq. Agreed `nyquist` by itself is not great because it's not immediately clear you're not specifying the Nyquist rate or Nyquist freq of the data.

Comment (AtsushiSakai) [reply scipy/signal/filter_design.py]: Thank you for review. I fixed the code based on your reviews.

File: scipy/signal/filter_design.py
Comment (AtsushiSakai) [reply scipy/signal/filter_design.py]: @larsoner  I fixed the code based on your reviews.

File: scipy/signal/tests/test_filter_design.py
Comment (AtsushiSakai) [reply scipy/signal/tests/test_filter_design.py]: I added. Please check the latest code.

**Review by larsoner — APPROVED** [review]
LGTM +1 for merge. But it would be good to get one more approval before merge (e.g., to make sure people are happy with the naming)

File: scipy/signal/filter_design.py
Comment (e-q) [inline scipy/signal/filter_design.py]: For the sake of being explicit, you may want to add something like "and is otherwise ignored."

Comment (AtsushiSakai) [reply scipy/signal/filter_design.py]: @e-q Thank you for your review. I added a docstring  based on you comment.

Comment (larsoner) [reply scipy/signal/filter_design.py]: I think @e-q was talking about the `include_nyquist` variable being ignored under the conditions, rather than the Nyquist freq being ignored, i.e., doing at the end "... frequency (Nyquist frequency) and is otherwise ignored.".

If you really want to say something about the `False` condition, it would be better to say that it's "omitted" rather than "ignored", but I don't think we need to since it's implied.

Comment (AtsushiSakai) [reply scipy/signal/filter_design.py]: Oh, OK. I understand. I fixed the docstring.

**Comment by larsoner** [comment]
Travis error is unrelated, I'll go ahead and merge. Thanks @AtsushiSakai

**Comment by ilayn** [comment]
Thanks all and sorry for the delay!

**Comment by mdhaber** [comment]
Thanks @AtsushiSakai! Could you add this contribution to the 1.5.0 [release notes](https://github.com/scipy/scipy/wiki/*Release-note-entries-for-SciPy-1.5.0*)?

**Comment by AtsushiSakai** [comment]
@mdhaber Hi. I added it. Please check it.

**Comment by mdhaber** [comment]
Thanks!

**Pull request merged**
